superduperdb/backends/mongodb/utils.py

```diff
@@ -0,0 +1,57 @@
+from pymongo import MongoClient
+from pymongo.errors import ServerSelectionTimeoutError
+
+from superduperdb import logging
+from superduperdb.misc.anonymize import anonymize_url
+
+
+def get_avaliable_conn(uri: str, **kwargs):
+    """
+    Get an available connection to the database.
+    This can avoid some issues with database permission verification.
+    1. Try to connect to the database with the given URI.
+    2. Try to connect to the database with the base URI without database name.
+
+    :param uri: The URI of the database.
+    """
+    base_uri, db_name = uri.rsplit("/", 1)
+    kwargs.setdefault("serverSelectionTimeoutMS", 5000)
+
+    raise_e = None
+
+    # Try to connect to the database with the given URI
+    client: MongoClient = MongoClient(uri, **kwargs)
+    try:
+        client[db_name].list_collection_names()
+        return client
+    except ServerSelectionTimeoutError as e:
+        # If the server is not available, raise the exception
+        raise e
+    except Exception as e:
+        uri_mask = anonymize_url(uri)
+        logging.warn(
+            f"Failed to connect to the database with the given URI: {uri_mask}"
+        )
+        logging.error(str(e))
```

Comment:
should we warn here since we are trying to reconnect with base uri eventually down below?

Reply:
Yes, But I use info here in line 43-45